### ISSUE TITLE
Monitoring - steve blocks monitoring from finishing installation 

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#25741]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
- run rancher:mater-head 3623201ad single install
- add a cluster 
- enable the cluster monitoring 

**Result:**
- both the app `cluster-monitoring` and `monitoring-operator` are stuck in installing state 

<img width="1400" alt="Screen Shot 2020-03-03 at 1 05 49 PM" src="https://user-images.githubusercontent.com/6218999/75817628-b3005180-5d54-11ea-9ac9-f22823180f08.png">
<img width="1455" alt="Screen Shot 2020-03-03 at 1 33 58 PM" src="https://user-images.githubusercontent.com/6218999/75817955-4a65a480-5d55-11ea-9143-373e8963b7e6.png">
<img width="1429" alt="Screen Shot 2020-03-03 at 1 34 08 PM" src="https://user-images.githubusercontent.com/6218999/75817960-4b96d180-5d55-11ea-89e6-3077fa214076.png">

- Metrics show on the cluster dashboard, node page, etc, but the Grafana and Prometheus links are broken

**Other details that may be helpful:**
- disabling steve fixes the problem partially: Grafana and Prometheus are accessible, but apps show new errors
<img width="1413" alt="Screen Shot 2020-03-03 at 1 06 02 PM" src="https://user-images.githubusercontent.com/6218999/75817632-b4317e80-5d54-11ea-918a-d6f31f7bb248.png">
<img width="1419" alt="Screen Shot 2020-03-03 at 1 06 12 PM" src="https://user-images.githubusercontent.com/6218999/75817637-b562ab80-5d54-11ea-95fc-68c7e8dedcf7.png">

Rancher logs:
```
2020/03/03 20:05:24 [ERROR] AppController p-tjr5h/cluster-monitoring [helm-controller] failed with : failed to install app cluster-monitoring. Error: UPGRADE FAILED: a release named cluster-monitoring is in use, cannot re-use a name that is still in use
```

**Comment by jiaqiluo** [comment]
New findings on rancher:master-head 17a40f502

- the cluster monitoring apps are deployed successfully
- but links to Grafana and Prometheus still not work
- after disabling the cluster monitoring, apps are stuck in uninstalling state 

<img width="1430" alt="Screen Shot 2020-03-04 at 4 28 35 PM" src="https://user-images.githubusercontent.com/6218999/75933304-e1eef400-5e36-11ea-839f-6bbfe5c051e2.png">
<img width="1390" alt="Screen Shot 2020-03-04 at 4 31 40 PM" src="https://user-images.githubusercontent.com/6218999/75933307-e2878a80-5e36-11ea-8355-6cea034dd5d9.png">
<img width="1409" alt="Screen Shot 2020-03-04 at 4 31 47 PM" src="https://user-images.githubusercontent.com/6218999/75933308-e3202100-5e36-11ea-9a7b-3fa15b590268.png">

**Comment by cbron** [comment]
Similar issue to https://github.com/rancher/rancher/issues/25742
There are jobs (`operator-init-cluster-monitoring`/`operator-init-monitoring-operator`) in downstream cluster that are completed but helm is not recognizing that and fails to remove them and mark app as deployed. Starting to think this has something to do with watchers/informers.

Ex helm logs:
```
Watching for changes to Job operator-init-cluster-monitoring with timeout of 0s
Add/Modify event for operator-init-cluster-monitoring: ADDED
operator-init-cluster-monitoring: Jobs active: 1, jobs failed: 0, jobs succeeded: 0
```
You can see the helm/tiller processes for those just hanging: 
```
USER       PID  START   TIME COMMAND
root       735  17:52   0:00 rancher-helm upgrade --install --namespace cattle-prometheus monitoring-operator --timeout 0 --set operator.apiGroup=monitoring.coreos.
root       736  17:52   0:02 rancher-tiller --listen :42962 --probe-listen :58626

root      1231  18:21   0:01 rancher-tiller --listen :60924 --probe-listen :43186
root      1232  18:21   0:00 rancher-helm upgrade --install --namespace cattle-prometheus monitoring-operator --timeout 0 --set operator.resources.limits.memory=500

root      1246  18:21   0:01 rancher-tiller --listen :33356 --probe-listen :35169
root      1252  18:21   0:00 rancher-helm upgrade --install --namespace cattle-prometheus cluster-monitoring --timeout 0 --set exporter-kube-etcd.enabled=true,prome
```
When I try to do a manual status on the cluster-monitoring install it works but says pending, however when I do one of the monitoring-operator (`HELM_HOST=:42962 rancher-helm status monitoring-operator`) I get this error from tiller: ` [storage/driver] 2020/03/09 18:35:50 query: failed to query with labels: the server is currently unable to handle the request (get configmaps)` 


The following errors have shown up in logs, not directly around the install but at diff times, potentially unrelated: 
* `retrywatcher.go:129 Watch failed: the server is currently unable to handle the request (get jobs.batch)`
* `reflector.go:326] github.com/rancher/steve/pkg/clustercache/controller.go:184: watch of *v1.PartialObjectMetadata ended with: too old resource version:`

**Comment by jiaqiluo** [comment]
the fix is validated on master-head c34bdc2bb

the steps are the same. Monitoring is working as expected.
